app-android/src/main/java/com/ustadmobile/port/android/view/ReportFilterEditFragment.kt

```diff
@@ -272,4 +294,173 @@ class ReportFilterEditFragment : UstadEditFragment<ReportFilter>(), ReportFilter
+) {
+    LazyColumn(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp)
+    )  {
+
+        item {
+            UstadMessageIdOptionExposedDropDownMenuField(
+                value = uiState.reportFilter?.reportFilterField ?: 0,
+                label = stringResource(R.string.report_filter_edit_field),
+                options = FieldConstants.FIELD_MESSAGE_IDS,
+                error = uiState.fieldError,
+                enabled = uiState.fieldsEnabled,
+                onOptionSelected = {
+                    onReportFilterChanged(uiState.reportFilter?.shallowCopy{
+                        reportFilterField = it.value
+                    })
+                },
+            )
+        }
+
+        item {
+            Row {
+                UstadMessageIdOptionExposedDropDownMenuField(
+                    modifier = Modifier.weight(0.5F),
+                    value = uiState.reportFilter?.reportFilterCondition ?: 0,
+                    label = stringResource(R.string.report_filter_edit_condition),
+                    options = ConditionConstants.CONDITION_MESSAGE_IDS,
+                    error = uiState.conditionsError,
+                    enabled = uiState.fieldsEnabled,
+                    onOptionSelected = {
+                        onReportFilterChanged(uiState.reportFilter?.shallowCopy{
+                            reportFilterCondition = it.value
+                        })
+                    },
+                )
+
+                 Spacer(modifier = Modifier.width(10.dp))
+
+                UstadMessageIdOptionExposedDropDownMenuField(
+                    modifier = Modifier.weight(0.5F),
+                    value = uiState.reportFilter?.reportFilterDropDownValue ?: 0,
+                    label = stringResource(R.string.report_filter_edit_values),
+                    options = ContentCompletionStatusConstants.CONTENT_COMPLETION_STATUS_MESSAGE_IDS,
+                    error = uiState.valuesError,
+                    enabled = uiState.fieldsEnabled,
+                    onOptionSelected = {
+                        onReportFilterChanged(uiState.reportFilter?.shallowCopy{
+                            reportFilterDropDownValue = it.value
+                        })
+                    },
+                )
+            }
+        }
+
+        item {
+            UstadTextEditField(
+                value = uiState.reportFilter?.reportFilterValue ?: "",
+                label = stringResource(id = R.string.report_filter_edit_values),
+                error = uiState.valuesError,
+                enabled = uiState.fieldsEnabled,
+                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+                onValueChange = {
+                    onReportFilterChanged(uiState.reportFilter?.shallowCopy{
+                        reportFilterValue = it
+                    })
+                }
+            )
+        }
+
+        item {
```

Comment:
Please add a reportFilterBetweenValueVisible field on UiState. Then this should only visible when reportFilterBetweenValueVisible is true.